src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ImportStatusBauService.cs

```diff
@@ -49,34 +52,52 @@ public class ImportStatusBauService : IImportStatusBauService
                     return queryResults
                         .OrderByDescending(result => result.Timestamp)
                         .Select(async result =>
-                    {
-                        var releaseId = new Guid(result.PartitionKey);
-                        var dataFileName = result.RowKey;
-                        var release =
-                            await _contentDbContext.Releases
-                                .Where(r => r.Id == releaseId)
-                                .Include(r => r.Publication)
-                                .FirstAsync();
+                        {
+                            var releaseId = new Guid(result.PartitionKey);
+                            var dataFileName = result.RowKey;
 
-                        var releaseFileRef = await _contentDbContext.ReleaseFileReferences.FirstAsync(r =>
-                            r.ReleaseId == releaseId &&
-                            r.Filename == dataFileName &&
-                            r.ReleaseFileType == ReleaseFileTypes.Data);
+                            var releaseFileRef = await _contentDbContext.ReleaseFileReferences.Where(
+                                    r =>
+                                        r.ReleaseId == releaseId &&
+                                        r.Filename == dataFileName &&
+                                        r.ReleaseFileType == ReleaseFileTypes.Data)
+                                .Include(rfr => rfr.Release)
+                                .ThenInclude(r => r.Publication)
+                                .FirstOrDefaultAsync();
 
-                        return new ImportStatusBau()
-                        {
-                            SubjectTitle = null, // EES-1655
-                            SubjectId = releaseFileRef.SubjectId,
-                            PublicationId = release.PublicationId,
-                            PublicationTitle = release.Publication.Title,
-                            ReleaseId = release.Id,
-                            ReleaseTitle = release.Title,
-                            DataFileName = result.RowKey,
-                            NumberOfRows = result.NumberOfRows,
-                            Status = result.Status,
-                            StagePercentageComplete = result.PercentageComplete,
-                        };
-                    }).Select(t => t.Result).ToList();
+                            if (releaseFileRef == null)
+                            {
+                                _logger.LogWarning(
+                                    "No ReleaseFileReference found! " +
+                                    $"releaseId: \"{releaseId}\" " +
+                                    $"dataFileName: \"{dataFileName}\"");
+                                return null;
+                            }
+
+                            if (releaseFileRef.Release == null)
```

Comment:
Don't think we need this check as it shouldn't be possible for a `ReleaseFileReference` to exist without a `Release` as there's a foreign key between the two. If you're genuinely concerned there are `ReleaseFileReference`s without releases, then I would suggest checking prod to make sure that we don't have any that are orphaned like that.